src/subscription_manager/gui/registergui.py

```diff
@@ -1007,7 +1090,28 @@ def pre(self):
                  (self.info.get_property('owner-key'),
                   self.info.get_property('environment'))
         self.info.set_property('register-status', msg)
+        self.async.register_consumer(self.info.get_property('consumername'),
+                                     self.facts,
+                                     self.info.get_property('owner-key'),
+                                     self.info.get_property('environment'),
+                                     self.info.get_property('activation-keys'),
+                                     self._on_registration_finished_cb)
+
+        return True
 
+
+class PerformForceRegisterScreen(PerformRegisterScreen):
+
+    def __init__(self, reg_info, async_backend, facts, parent_window):
+        super(PerformForceRegisterScreen, self).__init__(reg_info, async_backend, facts, parent_window)
+
+    def _on_unregistration_finished_cb(self, retval, error=None):
```

Comment:
If RegisterWidget always has a PerformUnregisterScreen and a PerformRegisterScreen, I think we can avoid queuing a new async request from this callback. I think the code as is likely works, it's a little weird (though
service level async stuff does something similar...).

Having these calls nested makes the exception handling more confusing as well.  The pre() here could
potentially see exceptions from unregister and register (or PerformRegisterScreen._on_registration_finished_cb could end up emitting 'register-error' with Exceptions from register or unregister).